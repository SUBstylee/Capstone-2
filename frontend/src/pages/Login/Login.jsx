import { useState,useEffect } from "react";
import { Link,useNavigate,useLocation } from "react-router-dom";
import {Form,Button,Row,Col} from 'react-bootstrap';
import { useDispatch,useSelector } from "react-redux";
import Message from "../../components/Message/Message";
import Loader from '../../components/Loader/Loader';
import CustomInput from "../../components/CustomInput/CustomInput";
import {login} from '../../actions/userActions';
import MetaWrapper from "../../components/MetaWrapper/MetaWrapper";

import FormContainer from "../../components/FormContainer/FormContainer";

const Login = () => {
    const [email,setEmail]=useState('');
    const [password,setPassword]=useState('');

    const navigate = useNavigate();
    const location = useLocation();
    const dispatch=useDispatch();
    const userLogin=useSelector((state)=>state.userLogin);
    const {loading,error,userInfo}=userLogin;
    const redirect = location.search ? location.search.split('=')[1] : '/';

    useEffect(()=>{
        if(userInfo){
            navigate(redirect);
        };
    },[navigate,userInfo,redirect]);
    
    const submitHandler=(e)=>{
        e.preventDefault();
        dispatch(login(email, password));
    };

  return (
    <div className="Login">
        <MetaWrapper title='TAA-Login'/>
        <FormContainer>
            <h1>Sign In</h1>
            {error&&<Message variant='danger'>{error}</Message>}
            {loading&&<Loader/>}
            <Form onSubmit={submitHandler}>
                <CustomInput id='email' type='email' autoComplete='current-email' name='email' labelText='Email Address' required={true} value={email} onChange={(e)=>setEmail(e.target.value)}/>
                <CustomInput id='password' type='password' autoComplete='current-password' name='password' labelText='Password' required={true} value={password} onChange={(e)=>setPassword(e.target.value)}/>
                <Button className="my-2" type='submit' variant="primary">
                    Sign In
                </Button>    
            </Form>
            <Row className="py-3">
                <Col>
                    New Customer? <Link to={redirect ? `/register?redirect=${redirect}`:'/register'}>Register</Link>
                </Col>    
            </Row>
        </FormContainer>
    </div>
  );
};

export default Login;