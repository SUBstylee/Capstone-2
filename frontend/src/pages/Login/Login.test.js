import { render } from '@testing-library/react';
import Login from './Login';
import store from '../../store';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

test("renders Login component without crashing", () => {
    const { getByText } = render(
        <Provider store={store}><BrowserRouter><Login /></BrowserRouter></Provider>
    );

    expect(getByText("Sign In")).toBeInTheDocument();
});

test("matches snapshot", function () {
    const { asFragment } = render(
        <Provider store={store}><BrowserRouter><Login /></BrowserRouter></Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});