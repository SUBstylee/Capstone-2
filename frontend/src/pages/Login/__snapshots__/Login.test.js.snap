// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Login"
  >
    <div
      class="FormContainer"
    >
      <div
        class="container"
      >
        <div
          class="justify-content-md-center row"
        >
          <div
            class="col-md-6 col-12"
          >
            <h1>
              Sign In
            </h1>
            <form
              class=""
            >
              <div
                class="CustomInput"
              >
                <input
                  autocomplete="current-email"
                  id="email"
                  name="email"
                  required=""
                  type="email"
                  value=""
                />
                <label
                  for="email"
                >
                  <span
                    style="transition-delay: 0ms;"
                  >
                    E
                  </span>
                  <span
                    style="transition-delay: 50ms;"
                  >
                    m
                  </span>
                  <span
                    style="transition-delay: 100ms;"
                  >
                    a
                  </span>
                  <span
                    style="transition-delay: 150ms;"
                  >
                    i
                  </span>
                  <span
                    style="transition-delay: 200ms;"
                  >
                    l
                  </span>
                  <span
                    style="transition-delay: 250ms;"
                  >
                     
                  </span>
                  <span
                    style="transition-delay: 300ms;"
                  >
                    A
                  </span>
                  <span
                    style="transition-delay: 350ms;"
                  >
                    d
                  </span>
                  <span
                    style="transition-delay: 400ms;"
                  >
                    d
                  </span>
                  <span
                    style="transition-delay: 450ms;"
                  >
                    r
                  </span>
                  <span
                    style="transition-delay: 500ms;"
                  >
                    e
                  </span>
                  <span
                    style="transition-delay: 550ms;"
                  >
                    s
                  </span>
                  <span
                    style="transition-delay: 600ms;"
                  >
                    s
                  </span>
                </label>
              </div>
              <div
                class="CustomInput"
              >
                <input
                  autocomplete="current-password"
                  id="password"
                  name="password"
                  required=""
                  type="password"
                  value=""
                />
                <label
                  for="password"
                >
                  <span
                    style="transition-delay: 0ms;"
                  >
                    P
                  </span>
                  <span
                    style="transition-delay: 50ms;"
                  >
                    a
                  </span>
                  <span
                    style="transition-delay: 100ms;"
                  >
                    s
                  </span>
                  <span
                    style="transition-delay: 150ms;"
                  >
                    s
                  </span>
                  <span
                    style="transition-delay: 200ms;"
                  >
                    w
                  </span>
                  <span
                    style="transition-delay: 250ms;"
                  >
                    o
                  </span>
                  <span
                    style="transition-delay: 300ms;"
                  >
                    r
                  </span>
                  <span
                    style="transition-delay: 350ms;"
                  >
                    d
                  </span>
                </label>
              </div>
              <button
                class="my-2 btn btn-primary"
                type="submit"
              >
                Sign In
              </button>
            </form>
            <div
              class="py-3 row"
            >
              <div
                class="col"
              >
                New Customer? 
                <a
                  href="/register?redirect=/"
                >
                  Register
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
